package.json

```diff
@@ -38,6 +38,7 @@
   "author": "Microsoft Conversation Learner Team",
   "license": "MIT",
   "dependencies": {
+    "@azure/cosmos": "^3.3.6",
```

Comment:
If I saw correctly this package was only used by the `CosmosLogStorage` as sample implementation of `ILogStorage`. 

1. This sample could be isolated so the SDK doesn't look like it depends on cosmos. 
2. Related but both of these types are part of the SDK which creates catch 22 dependency situation. You need SDK to implement `ILogStorage` but your package only needs the type and not the whole development kit with UI, router, etc.

Reply:
Merits further discussion.   My feeling is that it's only a very small subset of users that would choose to implement their own customer non-cosmos storage.  Makes the system extensible, but it feels pretty core to me

Reply:
probably better to move the implementation to another package. for example botbuilder has botbuilde-core and botbuilder-storage. we can use the same pattern here. 

---
In reply to: [344365542](https://github.com/microsoft/ConversationLearner-SDK/pull/512#discussion_r344365542) [](ancestors = 344365542)

Reply:
That means creating yet another npm package.